samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,279 @@
+        containing the AWS Resource Type and the list of IAM Action permissions generated for that resource.
+    """
+
+    partition_ex = r"(aws|aws-cn|aws-us-gov)"
+    region_ex = r"(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d"
+    account_ex = r"\d{12}"
+
+    default_permissions_map = {
+        rf"^arn:{partition_ex}:lambda:{region_ex}:{account_ex}:function:[\w-]+(:\d+)?$": {
+            "Resource": "AWS::Lambda::Function",
+            "Action": ["lambda:InvokeFunction"],
+        },
+        rf"^arn:{partition_ex}:apigateway:{region_ex}::\/apis\/\w+$": {
+            "Resource": "AWS::ApiGateway::Api",
+            "Action": ["execute-api:Invoke"],
+        },
+        rf"^arn:{partition_ex}:apigateway:{region_ex}::\/restapis\/\w+$": {
+            "Resource": "AWS::ApiGateway::RestApi",
+            "Action": ["execute-api:Invoke"],
+        },
+        rf"^arn:{partition_ex}:sqs:{region_ex}:{account_ex}:[\w-]+$": {
+            "Resource": "AWS::SQS::Queue",
+            "Action": ["sqs:SendMessage"],
+        },
+        rf"^arn:{partition_ex}:s3:::[\w-]+$": {
+            "Resource": "AWS::S3::Bucket",
+            "Action": ["s3:PutObject", "s3:GetObject"],
+        },
+        rf"^arn:{partition_ex}:dynamodb:{region_ex}:{account_ex}:table\/[\w-]+$": {
+            "Resource": "AWS::Dynamodb::Table",
+            "Action": [
+                "dynamodb:GetItem",
+                "dynamodb:PutItem",
+            ],
+        },
+        rf"^arn:{partition_ex}:states:{region_ex}:{account_ex}:stateMachine:[\w-]+$": {
+            "Resource": "AWS::StepFunctions::StateMachine",
+            "Action": [
+                "stepfunction:StartExecution",
+                "stepfunction:StopExecution",
+            ],
+        },
+    }
+    return default_permissions_map
+
+
+def _get_action_mapping(resource_arn: str) -> Union[dict, None]:
+
+    """
+    Returns a dictionary mapping a Resource type to a list of IAM actions generated for that resource.
+
+    Returns `None` if the supplied ARN is for a Resource type without any IAM actions supported.
+
+    Parameters
+    ----------
+    resource_arn:
+        The arn of the resource for which the mapping is returned.
+
+    Returns
+    -------
+    dict:
+        A dictionary containing two keys: `'Resource'` and `'Action'`. `'Resource'` contains the AWS Resource for which the IAM actions list contained in `'Action'` maps to.
+
+        For example `{"Resource": "AWS::S3::Bucket", "Action": ["s3:PutObject", "s3:GetObject"]}`
+
+    """
+
+    permissions_map = _get_permissions_map()
+
+    for arn_exp, mapping in permissions_map.items():
+        if re.search(arn_exp, resource_arn) is not None:
+            LOG.info("Matched ARN `%s` to IAM actions %s", resource_arn, mapping["Action"])
+            return mapping
+
+    # If the supplied ARN is for a resource without any default actions supported
```

Comment:
Since we log in the case where we find a match, is it worth logging when we fail to?

Reply:
I had actually done this, not sure why I removed it :/